AdyenCard/Components/Card/CardPublicKeyFetcherProtocol.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import Foundation
+
+internal protocol CardPublicKeyFetcherProtocol: PaymentComponent {
+    var cardPublicKeyProvider: AnyCardPublicKeyProvider { get set }
+}
+
+extension CardPublicKeyFetcherProtocol {
+    internal typealias CardKeySuccessHandler = (_ cardPublicKey: String) -> Void
+    internal typealias CardKeyFailureHandler = (_ error: Swift.Error) -> Void
+
+    internal func fetchCardPublicKey(onError: CardKeyFailureHandler? = nil, completion: @escaping CardKeySuccessHandler) {
+        do {
+            try cardPublicKeyProvider.fetch { [weak self] in
```

Comment:
same here with 2 types of success reported, either in completion handler as a `Result`, or throwing an `Error`, which is handled the same way as `Result.Failure`